platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -62,71 +68,53 @@ A new method will be added to the platform state to allow the App to submit a Ca
 void setCandidateRoster(@NonNull final Roster candidateRoster);
 ```
 
+The steps to validate and add the Roster to the state will be executed within the same thread that submits the request.
+An exception will be thrown if the Roster is invalid.
+The durability of the submitted Roster is ensured via storing it in the state.
+
 The existing `PlatformState` class is in the middle of a refactor, and in the long term, may cease to exist.
 However, its replacement will continue to have this new API to set a candidate roster.
 
 ## Data Structure
-The Hedera app is already responsible for managing the address book.
+
+The App is already responsible for managing the address book.
 We propose that it continues to do so, and when it receives a HAPI transaction, creates a candidate Roster object
 and passes it to the platform via the new API. The platform will then validate the roster and store it in the state.
 
-A Roster has a data structure as follows:
-
-```
-class Roster {
-    List~RosterEntry~ entries
-}
-
-class RosterEntry {
-    NodeId nodeId
-    long weight
-    X509Certificate gossipCaCertificate
-    PairingPublicKey tssEncryptionKey
-    List~ServiceEndpoint~ gossipEndpoints
-}
-
-class ServiceEndpoint {
-    bytes ipAddressV4
-    int32 port
-    string domain_name
-}
-
-Roster "1" *-- "many" RosterEntry
-RosterEntry "1" *-- "many" ServiceEndpoint
-
-```
-
 ### Protobuf
 
 #### Roster
+
 The protobuf definition for the Roster will look as follows:
 
 ```proto
 /**
  * A single roster in the network state.
  * <p>
- * The roster SHALL be a list of `RosterEntry` objects.
+ * The roster SHALL be a PBJ map of key `NodeId`, value `RosterEntry`.
  */
 message Roster {
 
     /**
-     * List of roster entries, one per consensus node.
+     * Map of roster entries, one per consensus node.
      * <p>
-     * This list SHALL contain roster entries in natural order of ascending node ids.
-     * This list SHALL NOT be empty.<br/>
+     * This map SHALL contain roster entries in natural order of ascending node ids.
+     * The node Id key SHALL match the node Id in the value.
+     * This map SHALL NOT be empty.<br/>
      */
-    repeated RosterEntry rosters = 1;
+    map<uint64, RosterEntry> mapInt32ToString = 1;
```

Comment:
Minor: PBJ is just an implementation of the protobuf support. In theory, this protobuf definition can be used with other implementations, for example, with Google protoc. I'd nuke the word "PBJ" altogether here. But if you insist on having a qualifier, I'd suggest to replace it with "protobuf" instead.

Also, please change the name of the field to, say, `rosterEntries` or something alike.

Reply:
Thanks for fixing this. Looks great!